packages/client/src/object/Cache.test.ts

```diff
@@ -42,4 +70,347 @@ describe("AsyncCache", () => {
+          i++
+        ) {
+          if (asyncCacheSpies.get.mock.calls[i][1] === key) {
+            expect(asyncCacheSpies.get.mock.results[i].type).toBe(
+              "throw",
+            );
+            expect(asyncCacheSpies.get.mock.results[i].value)
+              .toMatchInlineSnapshot(`[Error: aError]`);
+          }
+        }
+      });
+    }
+
+    function itDoesNotRejectLastRequestRightAway() {
+      it("does not reject right away", () => {
+        expect(asyncCacheSpies.get.mock.results.at(-1)?.type).toBe(
+          "return",
+        );
+      });
+    }
+
+    function describeRejectsFactoryCall(
+      num: number,
+      key: string,
+      fn: () => void,
+    ) {
+      return describe(`rejects factory request ${num} for ${key}`, () => {
+        beforeEach(rejectFactoryCall(num, key));
+        itRejectsAllRequestsOf("a");
+
+        fn();
+      });
+    }
+
+    function describeResolvesFactoryCall(
+      num: number,
+      key: string,
+      fn: () => void,
+    ) {
+      return describe(`resolves factory request ${num} for ${key}`, () => {
+        beforeEach(resolveFactoryCall(num, key));
+        fn();
+      });
+    }
+
+    function describeAsyncCacheGetFor(key: string, fn: () => void) {
+      return describe(`gets a request for '${key}'`, () => {
+        beforeEach(initiateAsyncCacheGetFor(key));
+        itDoesNotRejectLastRequestRightAway();
+
+        fn();
+      });
+    }
+
+    function itFullfillsAsyncCacheGets(indexes: number[], value: string) {
+      it(`fullfills the AsyncCache.get()'s ${indexes.join(", ")} with '${value}'`, async () => {
+        for (const i of indexes) {
+          expect(await pStateAsync(getPromises[i])).toBe("fulfilled");
+          expect(await getPromises[i]).toBe(value);
+        }
+      });
+    }
+
+    function itLeavesAsyncGetPromisesInStates(states: PromiseState[]) {
+      it(`leaves the AsyncCache.get()'s in states [${states.join(", ")}]`, async () => {
+        expect(await Promise.all(getPromises.map(p => pStateAsync(p))))
+          .toStrictEqual(states);
+        for (let i = 0; i < states.length; i++) {
+          expect(await pStateAsync(getPromises[i])).toBe(states[i]);
+        }
+      });
+    }
+
+    function itHasOnlyInvokedFactoryNTimes(n: number) {
+      it(`has only invoked the factory ${n} times`, () => {
+        expect(factoryFn).toHaveBeenCalledTimes(n);
+      });
+    }
+
+    function itReturnsForAsyncGet(results: any[]) {
+      it("returns for async get", () => {
+        expect(asyncCacheSpies.get.mock.results.map(a => a.value))
+          .toEqual(results);
+      });
+    }
+
+    it("has a minimal base state", () => {
+      expect(cache.get).not.toHaveBeenCalled();
+      expect(inProgress.get).not.toHaveBeenCalled();
+      expect(factoryFn).not.toHaveBeenCalled();
+      expect(inProgress.set).not.toHaveBeenCalled();
+    });
+
+    describeAsyncCacheGetFor("a", () => {
```

Comment:
I think this block needs a bunch more awaits, right?

Reply:
The describes don't need awaits because their job is to registered the jobs. They aren't supposed to do anything else. The awaits belong in before/after/it's.